### PR TITLE
Move dblwhole to head.shape

### PR DESCRIPTION
**Pull request by rettinghaus**
This moves `dblwhole` back to `head.shape` and extends `data.NOTEHEADMODIFIER.pat` to allow solmization syllables. 

Compare the SMuFL ranges
https://w3c.github.io/smufl/latest/tables/note-name-noteheads.html
https://w3c.github.io/smufl/latest/tables/note-name-noteheads-supplement.html

### CODE REVIEW

**Review by bwbohl — APPROVED** [review]

**Comment by bwbohl** [comment]
@music-encoding/maintainers please review

**Comment by ahankinson** [comment]
Are there any issues describing these changes? Particularly the extension to the note heads?

**Comment by rettinghaus** [comment]
Allowing the solmization syllables as text in the notehead seemed to be missing, as pitch names were already allowed. 
Also this allows some semantics over using just SMuFL code points for these noteheads (which can also get quite messy regarding different durations).

**Comment by ahankinson** [comment]
I have so many questions -- not particularly on this change, but on this section of the spec in general:

1. What does `.pat` stand for? (Helpful for understanding what this is trying to do...)
2. What is the `centertext` bit at the beginning of the pattern? Is this a literal string, or is it some sort of formatting?
3. What types of values is this pattern checking: `(H(s|♯)`?

File: source/modules/MEI.xml
Comment (ahankinson) [inline source/modules/MEI.xml:3090]: Should this also allow “Ut”?

Comment (rettinghaus) [reply source/modules/MEI.xml:3090]: As far as I know it is not used in modern solmization (and haven't really been used since the 17th century). As written above I simply followed the corresponding SMuFL ranges.

**Comment by pe-ro** [comment]
> I have so many questions -- not particularly on this change, but on this section of the spec in general:
> 
>     1. What does `.pat` stand for? (Helpful for understanding what this is trying to do...)

data.NOTEHEADMODIFIER references 2 other datatypes: data.NOTEHEADMODIFER.list, which allows a closed list of values, and data.NOTEHEADMODIFIER.pat (where 'pat' is shorthand for 'pattern'), which allows values matching regex patterns.  The objective is to permit either a value from the closed list or a value that matches one of the patterns defined in data.NOTEHEADMODIFER.pat.

>     2. What is the `centertext` bit at the beginning of the pattern? Is this a literal string, or is it some sort of formatting?

It's a literal string. The value is expected to look like "centertext()", where the text that will appear in the notehead is inside the parentheses.  I'm not completely convinced it's necessary because the description of the datatype says "Captures text rendered in the center of the notehead."  Also, "centered-ness" is implied in the definition/description of data.NOTHEADMODIFIER.list.  It seems better to me to leave it to a renderer to decide whether the modifying symbol or text will actually be centered.

>     3. What types of values is this pattern checking: `(H(s|♯)`?

This allows "H", which is the designation for B natural in German-speaking regions.

File: source/modules/MEI.xml
Comment (pe-ro) [reply source/modules/MEI.xml:3090]: I think "Ut" should be allowed for use in pre-modern notation.

**Comment by pe-ro** [comment]
The solmization patterns should allow "Ut", in my opinion.

**Comment by musicEnfanthen** [comment]
@ahankinson @pe-ro What's the status of this? Does this PR need more discussion, or is it safe to start a review?

**Comment by pe-ro** [comment]
It's fine with me to provide a double whole value in head.shape.  However, I don't recommend removing the value "dblwhole" from data.NOTEHEADMODIFIER.list.  I would rename it to "fences" instead.  The idea is to separate the head shape from the presence of the fences, thereby allowing any shape to have fences or not.

**Review by kepper — CHANGES_REQUESTED** [review]
during 2022 dev meeting in Berlin, @musicEnfanthen, @fujinaga, @pe-ro and @kepper came to the conclusion that this should have be done slightly differently. The new "dbwlwhole" value for `data.HEADSHAPE.list`would not be necessary, if instead we kept the "dblwhole" value for `data.NOTEHEADMODIFIER.list` and renamed it to "fences" instead. This means that you'd still have to use two separate attributes to get a whole note head with fences, but it would avoid a "shortcut" in the encoding, and would also avoid collision of having a "pre-built" fences combined with explicit "fences" as modifiers. The `.pat`stuff seems ok.

**Comment by musicEnfanthen** [comment]
@rettinghaus Any chance to rework this according to the proposal above?

**Comment by ahankinson** [comment]
Just because there is no reference to the original ticket (I think?) that motivated this change:

https://github.com/music-encoding/music-encoding/issues/865

(A further appeal to **please** reference issues in pull requests, otherwise we lose track of what we're trying to fix.)

**Comment by musicEnfanthen** [comment]
Actually, there was a reference to #865 in the original PR description (see edited version). Somehow it got lost, maybe because the discussion in 865 developed into a more broader direction? But be it as it may, thanks for the reminder and re-reference.

**Comment by ahankinson** [comment]
Could someone (@kepper?) provide an example of how you might encode a note with this plan?

**Comment by musicEnfanthen** [comment]
Pinging @rettinghaus and @bwbohl to add some additional comments.

**Comment by pe-ro** [comment]
MEI Development Meeting, 25/05/2023: Closing this pull request in favor of addressing #865 with a new pull request.

**Pull request closed without merging**
